### PR TITLE
Add integration tests for JaCoCo offline instrumented.

### PR DESCRIPTION
**Pull request by ZSmallX**
Notes that robolectric 4.9.x made changes on instrumenting logics, which may break JaCoCo instrument(on-the-fly or offline).

Add comparable tests for JaCoCo offline on custom shadows.

### CODE REVIEW

File: integration_tests/jacoco-offline/build.gradle [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+configurations {
+    jacocoAnt
+    jacocoRuntime
+}
+
+
+dependencies {
+    implementation "org.jacoco:org.jacoco.agent:${jacocoVersion}:runtime"
```
Comment (utzcoz) [inline integration_tests/jacoco-offline/build.gradle:34]: What about removing {} from these dependency names?

File: integration_tests/jacoco-offline/src/test/java/org/robolectric/integrationtests/mockk/Shadow.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package org.robolectric.integrationtests.mockk;
+
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.robolectric.integrationtests.jacoco.UnderShadow;
+
+@Implements(UnderShadow.class)
+public class Shadow {
```
Comment (utzcoz) [inline integration_tests/jacoco-offline/src/test/java/org/robolectric/integrationtests/mockk/Shadow.java:8]: I don't recommend you to use the class name `Shadow`. What about using name like `JaCoCoOfflineTester` and `ShadowJaCoCoOfflineTester` or other proper names?

File: integration_tests/jacoco-offline/build.gradle (hunk 1 above)
Comment (ZSmallX) [reply integration_tests/jacoco-offline/build.gradle]: yep "org.jacoco:org.jacoco.agent:${jacocoVersion}:runtime" -> "org.jacoco:org.jacoco.agent:$jacocoVersion:runtime"

File: integration_tests/jacoco-offline/src/test/java/org/robolectric/integrationtests/mockk/Shadow.java (hunk 2 above)
Comment (ZSmallX) [reply integration_tests/jacoco-offline/src/test/java/org/robolectric/integrationtests/mockk/Shadow.java]: Good idea! The name is good.

**Comment by utzcoz** [comment]
@ZSmallX You can close this PR, and continue your work https://github.com/robolectric/robolectric/pull/8016. If you like, you can split https://github.com/robolectric/robolectric/pull/8016 with two commmits, one for jacoco fixing and one for integration tests.

**Pull request closed without merging**